community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/LeafPlannerList.scala

```diff
@@ -23,8 +23,8 @@ import org.neo4j.cypher.internal.compiler.v2_2.planner.QueryGraph
 import org.neo4j.cypher.internal.compiler.v2_2.planner.logical.plans.LogicalPlan
 
 case class LeafPlannerList(leafPlanners: LeafPlanner*) {
-  def candidates(qg: QueryGraph)(implicit context: LogicalPlanningContext): Iterable[Seq[LogicalPlan]] = {
-    val LogicalPlans = leafPlanners.flatMap(_(qg))
-    LogicalPlans.groupBy(_.availableSymbols).values
+  def candidates(qg: QueryGraph, f: (LogicalPlan, QueryGraph) => LogicalPlan)(implicit context: LogicalPlanningContext): Iterable[Seq[LogicalPlan]] = {
+    val logicalPlans = leafPlanners.flatMap(_(qg)).map(f(_,qg))
+    logicalPlans.groupBy(_.availableSymbols.filterNot(_.name.endsWith("$$$_"))).values
```

Comment:
I know this is pretty horrible, but it is the only way I could imagine to fix this aggregation while we have this beauty in place: https://github.com/neo4j/neo4j/blob/master/community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/steps/projectEndpoints.scala#L64